tldraw/apps/tldraw-logseq/src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -0,0 +1,129 @@
+import { useApp } from '@tldraw/react'
+import { MOD_KEY} from '@tldraw/core'
+import { observer } from 'mobx-react-lite'
+import * as React from 'react'
+
+import * as ReactContextMenu from '@radix-ui/react-context-menu'
+
+const preventDefault = (e: Event) => e.stopPropagation()
+
+interface ContextMenuProps {
+  children: React.ReactNode
+}
+
+export const ContextMenu = observer(function ContextMenu({ children }: ContextMenuProps) {
+  const app = useApp()
+  const rContent = React.useRef<HTMLDivElement>(null)
+
+  return (
+    <ReactContextMenu.Root onOpenChange={state => {if (!state) app.transition('select')}}>
+      <ReactContextMenu.Trigger>{children}</ReactContextMenu.Trigger>
+      <ReactContextMenu.Portal>
```

Comment:
Do you think it is worth adding something similar to `ContextBarContainer` that renders the menu just like other elements on the canvas, other than using a portal?
Also since we have total control of the board including the state, I don't think the context menu component in Radix UI is necessary. We just need a menu that has regular accessibility. As a result, we do not need to worry about the issue of the Radix component you mentioned in the PR description.

Reply:
I tried to follow this paradigm on my first iteration. It ended up being an extremely verbose and complex implementation. I decided to switch to the context menu component when I realized that tldraw1.0 also uses the radix-ui context-menu component directly. I theory, controlling the state of the menu sounds like a better option. I could give it another shot here, or on a separate PR. Your call.

Reply:
No worries! If the current implementation works fine, we do not need to worry to about about that. After all the context menu does not have to follow the same path as the floating context bar, since the latter need to adapt its position to both the viewport and the currently selected shape.